homeassistant/components/bthome/binary_sensor.py

```diff
@@ -23,18 +26,124 @@
+    ),
+    BTHomeBinarySensorDeviceClass.LOCK: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.LOCK,
+        device_class=BinarySensorDeviceClass.LOCK,
+    ),
+    BTHomeBinarySensorDeviceClass.MOISTURE: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.MOISTURE,
+        device_class=BinarySensorDeviceClass.MOISTURE,
+    ),
+    BTHomeBinarySensorDeviceClass.MOTION: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.MOTION,
+        device_class=BinarySensorDeviceClass.MOTION,
+    ),
+    BTHomeBinarySensorDeviceClass.MOVING: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.MOVING,
+        device_class=BinarySensorDeviceClass.MOVING,
+    ),
+    BTHomeBinarySensorDeviceClass.OCCUPANCY: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.OCCUPANCY,
+        device_class=BinarySensorDeviceClass.OCCUPANCY,
+    ),
+    BTHomeBinarySensorDeviceClass.OPENING: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.OPENING,
+        device_class=BinarySensorDeviceClass.OPENING,
+    ),
+    BTHomeBinarySensorDeviceClass.PLUG: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.PLUG,
+        device_class=BinarySensorDeviceClass.PLUG,
+    ),
+    BTHomeBinarySensorDeviceClass.POWER: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.POWER,
+        device_class=BinarySensorDeviceClass.POWER,
+    ),
+    BTHomeBinarySensorDeviceClass.PRESENCE: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.PRESENCE,
+        device_class=BinarySensorDeviceClass.PRESENCE,
+    ),
+    BTHomeBinarySensorDeviceClass.PROBLEM: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.PROBLEM,
+        device_class=BinarySensorDeviceClass.PROBLEM,
+    ),
+    BTHomeBinarySensorDeviceClass.RUNNING: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.RUNNING,
+        device_class=BinarySensorDeviceClass.RUNNING,
+    ),
+    BTHomeBinarySensorDeviceClass.SAFETY: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.SAFETY,
+        device_class=BinarySensorDeviceClass.SAFETY,
+    ),
+    BTHomeBinarySensorDeviceClass.SMOKE: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.SMOKE,
+        device_class=BinarySensorDeviceClass.SMOKE,
+    ),
+    BTHomeBinarySensorDeviceClass.SOUND: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.SOUND,
+        device_class=BinarySensorDeviceClass.SOUND,
+    ),
+    BTHomeBinarySensorDeviceClass.TAMPER: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.TAMPER,
+        device_class=BinarySensorDeviceClass.TAMPER,
+    ),
+    BTHomeBinarySensorDeviceClass.UPDATE: BinarySensorEntityDescription(
+        key=BTHomeBinarySensorDeviceClass.UPDATE,
+        device_class=BinarySensorDeviceClass.UPDATE,
```

Comment:
When will this sensor be created? We prefer using update entities to represent firmware updates.

Reply:
As BTHome is a DIY format, a user could implement this functionality in his DIY sensor. I can add a note on bthome.io that this sensor is only intended to flag when a firmware update is available. Is that ok?

Reply:
I'd rather not support this binary sensor as we want it to be an update entity instead.